db/database_test.go

```diff
@@ -66,6 +66,11 @@ func setupTestDBForBucketWithOptions(t testing.TB, tBucket base.Bucket, dbcOptio
 
 func setupTestDBWithOptionsAndImport(t testing.TB, dbcOptions DatabaseContextOptions) *Database {
 	AddOptionsFromEnvironmentVariables(&dbcOptions)
+	if dbcOptions.GroupID == "" && base.IsEnterpriseEdition() {
+		// In a real SG this would be set in DefaultStartupConfig.
+		dbcOptions.GroupID = "default"
+		RegisterImportPindexImpl("default")
```

Comment:
This seems odd that it's not already done as part of `NewDatabaseContext` in the real SG code.
Having it done inside the function that generates an options struct (`dbcOptionsFromConfig`) feels a bit like a side-effect.

Reply:
Discussed offline - this should indeed be moved into `NewDatabaseContext`, but that's slightly outside the scope of this PR (and it's risky for a PR that's getting backported to 3.0.1), so I'll leave this as is with a TODO (to get the tests to pass) and file a separate one to move this out.